packages/compass-e2e-tests/helpers/commands/collection-workspaces.ts

```diff
@@ -12,7 +13,12 @@ async function navigateToCollection(
   // state of Schema, and Validation tabs without re-connecting.
   closeExistingTabs = true
 ): Promise<void> {
+  const connectionId = TEST_MULTIPLE_CONNECTIONS
+    ? await browser.getConnectionIdByName(connectionName)
+    : undefined;
```

Comment:
are all these checks necessary because we don't have the connectionName in single-connection, or because the getter function doesn't work for it?

Reply:
just wondering if we could avoid all those ifs somehow

Reply:
Actually I was confused and responded about the wrong thing at first. I thought of putting this check inside getConnectionIdByName, but then I have to make the return type be Promise<string|undefined> and then I have to check or assert the type of connectionId in a bunch of places.. this just felt cleaner overall.

We can't read out the connection id for single connection because it isn't somewhere persistent in the page.

Reply:
Actually I just tried it again and it didn't cause any problems. I think I just happened to change other things in the meantime.